docs/plugin-eval.md

```diff
@@ -0,0 +1,221 @@
+
+- Make sure your evaluation passes the `plugins` dictionary to the sampling command. This will ensure that the model is aware of your plugin and can use it during local evaluation.
+
+- It is best-practice to test the behavior of the model when your plugin is not installed. This helps effectively communicate the value of your plugin when combined with the model.
+
+- We've found evaluating the appropriateness of plugin use, the model's ability to recover from inappropriate or adversarial responses, and the ability for the model to integrate the plugin's knowledge into its own understanding of the world to be important evaluation dimensions.  You may find other dimensions to be important as well.
+
+## Extending Existing Plugins
+
+- Do not modify existing plugins. Instead, create a new plugin that extends the existing plugin using the plugin creation process above. When creating this new plugin, increase the version number of the plugin you are extending by 1.
+
+- Plugins can be forked by changing the namespace, but this should be done with care. We cannot allow two plugins with the same namespace to exist in practice.
+
+- Fixing bugs in existing plugins may be done if necessary, but please be thoughtful if the bug fix meaningfully changes the plugin's behavior. Not all plugins will be perfect, and evaluating our model's ability to recover from buggy behavior is an important part of the evaluation process.
+
+## Evaluation Workflows
+
+There are several factors to consider when assessing a model's performance using plugins.  We welcome additional contributions to this section, as it is not exhaustive.
+
+### Plugin State Population
+
+The method by which a plugin is populated with data is an important consideration when evaluating the model's behavior. There are two main ways to populate a plugin:
+
+#### Externally Populated
+
+In this scenario, the plugin is pre-populated with data outside of the current conversation. This allows the evaluation of the model's ability to rely on the plugin to obtain information about the state of the world. For example, a calendar plugin could be pre-populated with a meeting, and the model would then be asked to provide details about that meeting.
+
+Assuming a plugin named `MyCalendar` with `addMeeting` and `listMeetings` endpoints, the following input would be used to populate the plugin externally.  Setting the `include_in_conversation` flag to `false` will prevent the `addMeeting` action from being recorded in the conversation, allowing the plugin to act as if it was externally populated.  By default, this flag is `true`.
+
+```JSON
+{
+  "plugins": [
+    "openai.calendar.1",
+  ],
+  "ideal": [
+    "You have a meeting with the CEO at 2pm."
+  ],
+  "input": [
+    {
+      "role": "plugin",
+      "namespace": "MyCalendar",
+      "endpoint": "addMeeting",
+      "content": {"time": "2pm", "title": "Meeting with CEO"},
+      "include_in_conversation": false
```

Comment:
I'm a bit confused by what this actually does

Reply:
Adding the following:

```
Setting the `include_in_conversation` flag to `false` will prevent the `addMeeting` action from being recorded in the conversation, allowing the plugin to act as if it was externally populated.  By default, this flag is `true`.  In effect, this allows the plugin to be changed from a source distinct from the model during evaluation.
```